src/XrdHttp/XrdHttpProtocol.cc

```diff
@@ -1604,29 +1608,39 @@ void XrdHttpProtocol::Record() {
 
   int code = CurrentReq.getInitialStatusCode();
   if (code < 200) return;
-  auto duration = std::chrono::steady_clock::now() - CurrentReq.startTime;
+
+  XrdHttpMon::StatusCodes statusCode = XrdHttpMon::ToStatusCode(code);
+
+  std::chrono::steady_clock::duration duration{};
+  if (XrdHttpMon::hasGStream) {
+    duration = std::chrono::steady_clock::now() - CurrentReq.startTime;
+  }
 
   switch (CurrentReq.monState) {
     case XrdHttpReq::MonitState::NEW:
-      httpMon->RecordCount(CurrentReq.request, XrdHttpMon::ToStatusCode(code));
+      XrdHttpMon::RecordGStreamCount(CurrentReq.request, statusCode);
+      XrdHttpMon::RecordMonRollVerb(CurrentReq.request);
       CurrentReq.monState = XrdHttpReq::MonitState::ACTIVE;
       return;
 
     case XrdHttpReq::MonitState::ACTIVE:
-      httpMon->RecordSuccess(CurrentReq.request, XrdHttpMon::ToStatusCode(code), duration);
+      XrdHttpMon::RecordGStreamSuccess(CurrentReq.request, statusCode, duration);
+      XrdHttpMon::RecordMonRollStatus(statusCode);
       CurrentReq.monState = XrdHttpReq::MonitState::DONE;
       return;
 
     case XrdHttpReq::MonitState::ERR_NET:
-      httpMon->RecordErrNet(CurrentReq.request, XrdHttpMon::ToStatusCode(code), duration);
+      XrdHttpMon::RecordGStreamErrNet(CurrentReq.request, statusCode, duration);
+      XrdHttpMon::RecordMonRollStatus(statusCode);
       CurrentReq.monState = XrdHttpReq::MonitState::DONE;
       return;
 
     case XrdHttpReq::MonitState::ERR_PROT:
-      httpMon->RecordErrProt(CurrentReq.request, XrdHttpMon::ToStatusCode(code), duration);
+      XrdHttpMon::RecordGStreamErrProt(CurrentReq.request, statusCode, duration);
+      XrdHttpMon::RecordMonRollStatus(statusCode);
       CurrentReq.monState = XrdHttpReq::MonitState::DONE;
       return;
-    
+
     case XrdHttpReq::MonitState::DONE:
       eDest.Emsg("Record", "ERROR: Record called after state was set to DONE");
```

Comment:
Does this reset itself somehow or is it now in permanent error state?

Reply:
This state resets for each request. 

This should **Never** happen; if we see this in the logs ever, it would mean that I failed to capture the correct entry and exit points.